getting-started/installation/general-installation/installation-community-containerized.markdown

```diff
@@ -0,0 +1,285 @@
+---
+layout: default
+title: Installing Community Using Containers
+published: true
+sorting: 80
+---
+
+The instructions in this guide describe how to download and install the latest version of CFEngine Community in a Docker containerized environment using pre-compiled rpm packages and ubi9 images.
+
+This guide describes how to set up a client-server model with CFEngine and, through policy, manage both containers.
+
+Docker containers will be created, one container to be the Policy Server (server), and another container that will be the Host Agent (client).
+
+Both the containers will run **_ubi9-init_** images and communicate on a container network.
+Upon completion, you are ready to start working with CFEngine.
+
+
+## Requirements
+* 1G+ disk space
+* 1G+ memory
+* Working [Docker Engine](https://docs.docker.com/engine/) or [Podman](https://podman.io/) setups on a supported [x86_64](https://en.wikipedia.org/wiki/X86-64) platform.
+
+**Note**: This document considers [Docker Engine](https://docs.docker.com/engine/) for all examples.
+Use of [Podman](https://podman.io/) shall be similar with adequate adaptations. (_Ref_: [Emulating Docker CLI with Podman](https://podman-desktop.io/docs/migrating-from-docker/emulating-docker-cli-with-podman)).
+
+## Overview
+1. Installing container engine
+2. Preparing CFEngine hub in container
+3. Preparing CFEngine host in container
+4. Using docker compose
+    1. Preparing container image for CFEngine
+    2. Using docker compose service
+5. Glossary
+6. References
+
+## Installing container engine
+**Ref**: [Install Docker Engine](https://docs.docker.com/engine/install/)
+
+OR
+
+**Ref**: [Podman Installation Instructions](https://podman.io/docs/installation)
+(_Optionally_: [Emulating Docker CLI with Podman](https://podman-desktop.io/docs/migrating-from-docker/emulating-docker-cli-with-podman))
+
+## Preparing CFEngine hub in container
+Run the container with systemd
```

Comment:
In my experience running with systemd in a container on a Linux host can bring down your entire desktop environment at times when the container systemd bits are shutdown/restarted/etc.

Have you had this experience?

I will test your instructions soon. Would be great if this issue is not present (which I imagine it was not for you somehow, nice!).

Reply:
I haven't had any such experience. But, I can imagine that happening as I am using docker in `--privileged` mode, which is not ideal for Prod environments.

The idea for coming up with this document was when I wanted to have a easy playground for testing aspects of cfengine on a container (container being lean and it's not so easy to have a usable VM setup on a MacBook Air M2).
A reason for considering `ubi9-init` over a `debian` image was the need for systemd backbone for cfengine setup. `ubi9-init` was a quick go-to solution that could help me get started really quick.

The script for quick-install-cfengine-community.sh is broken for some reason
```
[root@8fa019968e7e /]# wget http://cfengine.package-repos.s3.amazonaws.com/quickinstall/quick-install-cfengine-community.sh
--2024-12-10 17:38:32--  http://cfengine.package-repos.s3.amazonaws.com/quickinstall/quick-install-cfengine-community.sh
Resolving cfengine.package-repos.s3.amazonaws.com (cfengine.package-repos.s3.amazonaws.com)... 54.231.196.169, 3.5.6.127, 52.217.227.233, ...
Connecting to cfengine.package-repos.s3.amazonaws.com (cfengine.package-repos.s3.amazonaws.com)|54.231.196.169|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3160 (3.1K) [text/x-shellscript]
Saving to: ‘quick-install-cfengine-community.sh’

quick-install-cfengine-community.sh  100%[=====================================================================>]   3.09K  --.-KB/s    in 0s       

2024-12-10 17:38:33 (124 MB/s) - ‘quick-install-cfengine-community.sh’ saved [3160/3160]

[root@8fa019968e7e /]# chmod a+x quick-install-cfengine-community.sh 
[root@8fa019968e7e /]# ./quick-install-cfengine-community.sh 
Sorry I was unable to determine the distro
[root@8fa019968e7e /]#
```
Let me check why this is happening and if I can fix it over this weekend.

Reply:
Ah, I see quick-install-cfengine-community referenced here: https://docs.cfengine.com/docs/3.24/getting-started-installation-general-installation-installation-community.html but I am not sure that is supported/updated so maybe something for us to fix.

I am going to push some commits here with re-work that I did using cf-remote. Seems good so far.

WRT to systemd troubles I was using ubuntu and `CMD [ "/lib/systemd/systemd" ]` which may have slightly different behavior! I will check on that. I have had no issues so far running these commands on my debian-12 system.